src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp

```diff
@@ -0,0 +1,180 @@
+ * 2J\frac{\partial \xi^{\hat{1}}}{\partial x^2}&=
+ * D^{(\hat{2})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{2})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{3})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{2}}}{\partial x^2}&=
+ * D^{(\hat{1})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{1})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{3})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{1}}}\right)\\
+ * 2J\frac{\partial \xi^{\hat{3}}}{\partial x^2}&=
+ * D^{(\hat{1})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{1})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{2}}}\right)
+ * +D^{(\hat{2})}\left(x^1\frac{\partial x^3}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{2})}\left(x^3\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{1}}}{\partial x^3}&=
+ * D^{(\hat{2})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{2})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{3})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{2}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{2}}}{\partial x^3}&=
+ * D^{(\hat{1})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{3}}}\right)
+ * -D^{(\hat{1})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{3}}}\right)
+ * +D^{(\hat{3})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{3})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right) \\
+ * 2J\frac{\partial \xi^{\hat{3}}}{\partial x^3}&=
+ * D^{(\hat{1})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{2}}}\right)
+ * -D^{(\hat{1})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{2}}}\right)
+ * +D^{(\hat{2})}\left(x^2\frac{\partial x^1}{\partial \xi^{\hat{1}}}\right)
+ * -D^{(\hat{2})}\left(x^1\frac{\partial x^2}{\partial \xi^{\hat{1}}}\right)
+ *
+ * \f}
+ *
+ * Again, this ensures that the metric identities are satisfied discretely. That
+ * is,
+ *
+ * \f{align*}{
+ *
+ * \partial_{\hat{\imath}}\left(J\frac{\partial\xi^{\hat{\imath}}}
+ * {\partial x^i}\right)=0
+ *
+ * \f}
+ *
+ * numerically.
+ *
+ * The reason for calculating \f$J\partial\xi^{\hat{\imath}}/\partial x^i\f$ in
+ * this manner is because in the weak form of DG (and most spectral-type methods
+ * can be recast as DG) we effectively evaluate
+ *
+ * \f{align*}{
+ *
+ * \partial_{\hat{\imath}}\left(J\frac{\partial\xi^{\hat{\imath}}}
+ * {\partial x^i} F^i\right),
+ *
+ * \f}
+ *
+ * which should be identically zero if \f$F^i\f$ is constant. This feature of a
+ * scheme is referred to as free-stream preserving. Another way to achieve
```

Comment:
Here I would say "Note that another way to..." to help make clear you're on an aside.